pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -78,6 +78,8 @@ type Registry struct {
 	// effectively be pulled from "example.com/foo/bar/myimage:latest".
 	// If no Prefix is specified, it defaults to the specified location.
 	Prefix string `toml:"prefix"`
+	// The credential store for the registry
+	CredHelper string `toml:"cred_helper"`
```

Comment:
If we are looking up credentials by registry host names, we should reject configurations that set this on inner namespaces.

Alternatively, we may need credentials per smaller namespaces (there’s a worrisome case of needing different Quay.io credentials for private repositories and for pulling OpenShift), but then `docker/config` API needs to be extended to accept full Docker references instead of just registry names.

Reply:
Initially, I thought this can match docker document, this configuration supports a specified registry domain. So to not set for an inner namespace, this configuration use substring before the first '/' of `Prefix` as the registry name?

Reply:
Just stripping namespaces from `Prefix` could result in ambiguities if there are different configurations for different namespaces of a single registry.

Instead, `postProcess` should validate the rules, and reject configurations that contain a `/`.